arch/arm64/src/bcm2711/bcm2711_i2c.c

```diff
@@ -77,6 +73,86 @@
 
 #define FIFO_DEPTH 16
 
+/* Logic for determining which alternate function should be used to achieve
+ * the user selected I2C pins for each interface
+ */
+
+#ifdef CONFIG_BCM2711_I2C0
+
+#if CONFIG_BCM2711_I2C0_SDA == 0 && CONFIG_BCM2711_I2C0_SCL == 1
+#define BCM2711_I2C0_ALT BCM_GPIO_FUNC0
+#elif CONFIG_BCM2711_I2C0_SDA == 28 && CONFIG_BCM2711_I2C0_SCL == 29
+#define BCM2711_I2C0_ALT BCM_GPIO_FUNC0
+#elif CONFIG_BCM2711_I2C0_SDA == 44 && CONFIG_BCM2711_I2C0_SCL == 45
+#define BCM2711_I2C0_ALT BCM_GPIO_FUNC1
+#elif CONFIG_BCM2711_I2C0_SDA == 46 && CONFIG_BCM2711_I2C0_SCL == 47
+#error "BCM2711 datasheet does not declare the alternate function needed for this pin pair."
+#else
+#error "Invalid I2C0 pin selections."
```

Comment:
Maybe include here the valid pins to help user selecting the pins:
#error "Invalid I2C0 pin selections, valid SDA/SCL pair pins combination: 0,1;  28,29; 44,45; 46,47"